### ISSUE TITLE
Multi-room Issue

### ISSUE THREAD

**Issue by cyberralf83** [homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum#242]
Hi, 
I am getting the following error when homekit starts: 
 "ERR getRoomList | roborock.vacuum.s4 | Could not find a timer for autoroom"

I have been using the "autoroom": ["Cafe", "Kitchen", "Dining Room", "Living room" ], with and omitting the "rooms" parameter.

I have set up the timer feature, with a 00:00 start time, selecting the rooms I want cleaned, and then turning it "off" when saved. 
Interestingly it also says "scheduled room cleaning will become invalid if room divided" on the set timing menu on the MiHome App. 

Any suggestions?

Thanks in advance

**Comment by afharo** [comment]
This feature was developed by @gothma. Can you help with answering this question?

**Comment by gothma** [comment]
Hi Ralf,

thanks for testing this! We did not try it our for an s4 so far, so there might be unforeseen issues. On reason might be that the s4 handles "rooms" as "zones" instead.

The "scheduled room cleaning will become invalid if room divided" is to my knowledge a default when using rooms in a timer. I can see it as well. As it says the timer (and thus the autoroom function) will stop to work when you change something on the room map.

From what I can read, you should have configured the timer correctly. I've checked the logging and I'm afraid a log won't help me. Since I don't have an s4, I might be able to assist you in debugging if you would be able to do that.

Best,
ziisch

**Comment by afharo** [comment]
Regarding the logs, can you run `homebridge` with the environment variable `DEBUG=thing:*` set? That will show the underlying messages sent to the robot and the responses. Maybe that can help us find out the issue :)

**Comment by gothma** [comment]
Thanks @afharo , I didn‘t know that. Yes a log with the robots response would help me a lot!

**Comment by enz1ey** [comment]
@gothma I'm also attempting to get rooms to work with my S4. So far I've got:

`Jul 05 14:34:31 PMS homebridge[3165]: [7/5/2020, 2:34:31 PM] [Living Room Vacuum] INF getCleaning | roborock.vacuum.s4 | Cleaning is false
Jul 05 14:34:31 PMS homebridge[3165]: [7/5/2020, 2:34:31 PM] [Living Room Vacuum] ACT setCleaning | roborock.vacuum.s4 | Start cleaning Room ID 1, not charging.
Jul 05 14:34:31 PMS homebridge[3165]: [7/5/2020, 2:34:31 PM] [Living Room Vacuum] ERR setCleaning | roborock.vacuum.s4 | Failed to set cleaning to true { Error: data for segment is not a number`

**Comment by enz1ey** [comment]
@gothma @cyberralf83 Scratch my last comment, looks like it's now working as of firmware version 3.5.8_0358!

I was previously on 3.5.7_0328 and noticed in the "Edit Room" screen, there was a "Name" field at the bottom, however attempting to name the rooms didn't seem to save anything. After updating to 3.5.8_0358, I tried naming the rooms once again and it saved to the map, so I set `"autoroom": true,` in the config and the homebridge logs output the correct room IDs just as it would with the S6. I then removed the `"autoroom": true,` line and added in the regular `"rooms": [...]` array and it works.
